torch_geometric/contrib/flag/flag.py

```diff
@@ -51,13 +51,54 @@ def __init__(
         self.device = device
         self.callbacks = callbacks
 
+    def _set_perturb(
+        self,
+        x: torch.Tensor,
+        train_idx: torch.Tensor,
+        unlabeled_idx: torch.Tensor,
+        step_size_labeled: float,
+        step_size_unlabeled: float,
+    ) -> None:
+        """
+        Set the perturb tensor by randomly initializing its values. Values
+        corresponding to both labeled and unlabeled samples are drawn from
+        uniform distributions, but these distributions may be parametrized
+        by different ranges.
+
+        Args:
+            x (torch.Tensor): Node feature matrix with shape
+                [num_nodes, num_node_features].
+            train_idx (torch.Tensor): Indices of training data.
+            unlabeled_idx (torch.Tensor): Indices of non-training data.
+            step_size_labeled (float): The step size to take during the
+                gradient ascent on the perturbation for labeled nodes. This
+                is also used as the ranges when initializing the perturbation.
+            step_size_unlabeled (float): The step size to take during the
+                gradient ascent on the perturbation for labeled nodes. This
+                is also used as the ranges when initializing the perturbation.
+        """
+        perturb_tensor = torch.zeros(*x.shape, dtype=torch.float)
+        labeled_values = torch.FloatTensor(len(train_idx),
```

Comment:
One issue, one recommendation here:
- The tensors created in this function need to be sent to `self.device` either by passing in via the `device` parameter or using `.to(self.device)`, otherwise it breaks when using the GPU
- `torch.FloatTensor(...)` and similar are [legacy constructors and so they've been deprecated](https://github.com/pytorch/pytorch/issues/28867#issuecomment-547823600)
    - It's recommended to use `torch.tensor(..., dtype=torch.float)` style instead
    - See [PyTorch types docs](https://pytorch.org/docs/stable/tensors.html) for `dtype` values that correspond to legacy constructors

Reply:
Thank you, will change!

Reply:
ChatGPT told me to use torch.FloatTensor :P I guess this is one of the few cases where it is wrong...